src/compile/miniconstants.rs

```diff
@@ -4,284 +4,85 @@
 
 //!Creates a fixed list of globally accessible constants.
 
+use crate::compile::CompileError;
 use crate::evm::{builtin_contract_path, AbiForContract};
 use crate::uint256::Uint256;
+use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+
+#[derive(Clone, Default, Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct ConstantsFile {
+    integer: Vec<(String, u64)>,
+    hex: Vec<(String, String)>,
```

Comment:
Is there a reason for using a `Vec` instead of a `HashMap` for this?

Reply:
This was done just as a way of mirroring the implementation from before, I agree a map like structure would be best here. However, I would prefer a `BTreeMap` over a `HashMap` in order to get deterministic iteration.

Reply:
That sounds even better haha. I don't know the rust standard library particularly well 😁